activerecord/lib/active_record/connection_adapters/postgresql/quoting.rb

```diff
@@ -130,13 +130,18 @@ def quote_default_expression(value, column) # :nodoc:
           end
         end
 
-        def type_cast(value) # :nodoc:
+        def type_cast(value, in_array: false) # :nodoc:
```

Comment:
What `in_array` means here? and why we need to pass it to this method, should not this information be in the value?